contract/lib.rs

```diff
@@ -0,0 +1,331 @@
+                points_by_sender: Mapping::default(),
+                senders: Vec::new(),
+                points_by_recipient: Mapping::default(),
+                recipients: Vec::new(),
+                total_points: 0,
+                max_points_per_sender: max_points_per_sender.unwrap_or(10),
+            }
+        }
+
+        #[ink(message)]
+        pub fn set_admin(&mut self, new_admin: AccountId) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            if !self.is_member(&new_admin) {
+                self.add_member(new_admin)?;
+            }
+
+            self.admin = new_admin;
+
+            self.env().emit_event(AdminChanged { admin: new_admin });
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn add_member(&mut self, new_member: AccountId) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            if self.members.contains(&new_member) {
+                return Err(ContractError::MemberAlreadyExists);
+            }
+
+            self.members.push(new_member);
+            self.env().emit_event(NewMember { member: new_member });
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn remove_member(&mut self, member_to_remove: AccountId) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            match self.members.iter().position(|m| *m == member_to_remove) {
+                Some(member_index) => {
+                    self.members.remove(member_index);
+                    self.env().emit_event(MemberDeleted {
+                        member: member_to_remove,
+                    });
+                }
+                None => return Err(ContractError::MemberDoesNotExist),
+            };
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn set_max_points_per_sender(
+            &mut self,
+            max_points_per_sender: u64,
+        ) -> Result<(), ContractError> {
+            self.check_admin()?;
+
+            if self.max_points_per_sender == 0 {
+                return Err(ContractError::MaxPointsPerSenderCannotBeZero);
+            }
+
+            self.max_points_per_sender = max_points_per_sender;
+
+            Ok(())
+        }
+
+        #[ink(message)]
+        pub fn award(&mut self, recipient: AccountId, amount: u64) -> Result<(), ContractError> {
+            let sender = self.env().caller();
+            self.validate_award_inputs(sender, recipient, amount)?;
+
+            let sender_used_points = self.points_by_sender.get(sender).unwrap_or(0);
+            self.validate_sender_points(sender_used_points, amount)?;
+            self.points_by_sender
+                .insert(sender, &(sender_used_points.checked_add(amount).unwrap()));
+
+            let recipient_points = self.points_by_recipient.get(recipient).unwrap_or(0);
+            self.points_by_recipient
+                .insert(recipient, &(recipient_points.checked_add(amount).unwrap()));
```

Comment:
To make it more explicit I would take out lines like:

`&(sender_used_points.checked_add(amount).unwrap())`
`&(recipient_points.checked_add(amount).unwrap())` and


To their own line where we name variables like "sender_used_points_result" and "recipients_points_result"